src/agent/src/image_rpc.rs

```diff
@@ -207,7 +207,15 @@ impl ImageService {
     ) -> Result<String> {
         info!(sl(), "image metadata: {:?}", image_metadata);
         Self::set_proxy_env_vars();
-        if image_metadata["io.kubernetes.cri.container-type"] == "sandbox" {
+        let is_sandbox = if let Some(value) = image_metadata.get("io.kubernetes.cri.container-type") {
+            value == "sandbox"
+        } else if let Some(value) = image_metadata.get("io.kubernetes.cri-o.ContainerType") {
```

Comment:
Personally, I'd prfer those two annotations being a const / enum, and looping into that to check whether value is sandbox.

OTOH, this branch is almost dead, so I wouldn't block the PR on this,